pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -369,15 +375,34 @@ func PublishDeviceInfoToZedCloud(ctx *zedagentContext) {
 					log.Errorf("error with get about RAID info from devTree: %v", err)
 				} else {
 					curentRaid = zfs.GetZpoolRaidType(vdevs)
+					for _, vdev := range vdevs.Devices {
+						// If this is a RAID or mirror, look at the disks it consists of
+						if vdev.Type == libzfs.VDevTypeMirror || vdev.Type == libzfs.VDevTypeRaidz {
+							for _, disk := range vdev.Devices {
+								rDiskStatus, err := zfs.GetZfsDiskAndStatus(disk)
+								if err == nil {
+									rStorageInfo.ZpoolState.Disks = append(rStorageInfo.ZpoolState.Disks, rDiskStatus)
+								}
+							}
+							break // for one pool, have one RAID
+						}
+						// If there is no RAID or mirror, add a disk if it is a disk
+						rDiskStatus, err := zfs.GetZfsDiskAndStatus(vdev)
+						if err == nil {
```

Comment:
Please log.Errorf if you get an error.

Reply:
Any update? Any reason to not log an error here? If so add a comment which helps the reader understand why errors need to be ignored.